### PR TITLE
Add cfg::Config._query_cache_mode

### LINKED ISSUE #7102: Add a temporary config to switch query cache mode
Before the end goal of proper persistent query cache is achieved with Postgres functions, we'd like to have a config to control the cache mode the server is running in, that is `Config._query_cache_mode`, which is an enum of:

* `IN_MEMORY` - query cache lives in memory and resets on server restart, no persistent cache is used
* `REG_INLINE` - query cache (the actual SQL) is placed inline in the cache registration table
* `PG_FUNC` - the actual SQL is encapsulated in Postgres functions and the cache registration table contains function calls
* `DEFAULT` - the distribution-default mode, currently `REG_INLINE` but it will be `PG_FUNC` in a future release

This will be a system config and server restart is needed for changes to be effective. If the effective config value changes on start, the cache registration table will be cleared.

This config is not permanent and will be dropped once the persistent cache system is stable.

### PR DESCRIPTION
**Pull request by fantix**
Now users may control the cache mode per statement or database. Changing the effective mode on a database will clear its cache registration table.

Fixes #7102, replaces #7122

### CODE REVIEW

File: edb/lib/cfg.edgeql [hunk 1]
```diff
@@ -192,6 +194,13 @@ ALTER TYPE cfg::AbstractConfig {
             'Recompile all cached queries on DDL if enabled.';
     };
 
+    CREATE PROPERTY _query_cache_mode -> cfg::QueryCacheMode {
```
Comment (msullivan) [inline edb/lib/cfg.edgeql:197]: Are we sure we want the leading `_`?

**Review by msullivan — APPROVED** [review]
Looks good!

File: edb/lib/cfg.edgeql (hunk 1 above)
Comment (fantix) [reply edb/lib/cfg.edgeql:197]: I guess so, like we'll probably drop this once cache is stable? Also I think it needs more work to be a public config, e.g. it's not yet clearing cache if changed on system level, or if we really want to allow all modes of cache exist at once by allowing per-statement cache policy (it works implicitly, just confusing).

Comment (msullivan) [reply edb/lib/cfg.edgeql:197]: Hm, ok.
I sort of expect we'll keep it, though?

Comment (fantix) [reply edb/lib/cfg.edgeql:197]: I'll drop the leading underscore - we just don't have to advertise it now.

**Pull request merged**